### PR TITLE
Make locales switchable by user & dependent on current channel

### PR DESCRIPTION
**Pull request by pamil**
| Q               | A
| --------------- | ---
| Bug fix?        | yes
| New feature?    | yes
| BC breaks?      | yes
| Related tickets | based on #5597 
| License         | MIT

### CODE REVIEW

File: src/Sylius/Behat/Context/Ui/Shop/LocaleContext.php [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+    }
+
+    /**
+     * @Then I should not be able to shop using the :localeName locale
+     */
+    public function iShouldNotBeAbleToShopUsingTheLocale($localeName)
+    {
+        $this->homePage->open();
+
+        if (in_array($localeName, $this->homePage->getAvailableLocales(), true)) {
```
Comment (Arminek) [inline src/Sylius/Behat/Context/Ui/Shop/LocaleContext.php]: Assert::true?

Comment (pamil) [inline src/Sylius/Behat/Context/Ui/Shop/LocaleContext.php]: There's no `Assert::notOneOf` and `Assert::true` / `Assert::false` seems weird.

File: src/Sylius/Bundle/CoreBundle/Resources/config/context.xml [hunk 2]
```diff
@@ -62,5 +62,33 @@
...
+            <argument type="service" id="sylius.context.channel" />
+            <argument>%locale%</argument>
+        </service>
+
+        <service id="sylius.storage.locale" class="Sylius\Component\Core\Locale\LocaleStorage">
+            <argument type="service" id="sylius.storage.session" />
+        </service>
+
+        <service id="sylius.context.locale.core" class="Sylius\Component\Locale\Context\CompositeLocaleContext" decorates="sylius.context.locale">
+            <call method="addContext">
```
Comment (Arminek) [inline src/Sylius/Bundle/CoreBundle/Resources/config/context.xml]: How about making an compiler pass for this. WDYT?

Comment (pamil) [inline src/Sylius/Bundle/CoreBundle/Resources/config/context.xml]: Might do, but I'd rather not make this PR bigger.

**Comment by pjedrzejewski** [comment]
Thank you Kamil! 👍

**Pull request merged**